examples/access_review/workflows/okta_password_policy_monitoring.py

```diff
@@ -0,0 +1,132 @@
+from typing import Annotated
+from datetime import datetime, timedelta
+import json
+
+from admyral.workflow import workflow, Schedule
+from admyral.typings import JsonValue
+from admyral.action import action, ArgumentMetadata
+from admyral.actions import get_okta_logs, send_slack_message_to_user_by_email
+
+
+@action(
+    display_name="Calculate Time Range for Last Full Hour",
+    display_namespace="Utilities",
+    description="Calculate the time range for the last full hour",
+)
+def calculate_time_range() -> tuple[str, str]:
+    end_time = datetime.utcnow().replace(minute=0, second=0, microsecond=0)
+
+    start_time = (end_time - timedelta(hours=1)).isoformat() + "Z"
+
+    return (start_time, end_time.isoformat() + "Z")
+
+
+@action(
+    display_name="Get Okta Password Policy Update Logs",
+    display_namespace="Okta",
+    description="Retrieve Okta password policy update logs for a specified time range",
+    secrets_placeholders=["OKTA_SECRET"],
+)
+def get_okta_password_policy_update_logs(
+    start_time: Annotated[
+        str,
+        ArgumentMetadata(
+            display_name="Start Time",
+            description="The start time for the logs to retrieve in ISO 8601 format",
+        ),
+    ],
+    end_time: Annotated[
+        str,
+        ArgumentMetadata(
+            display_name="End Time",
+            description="The end time for the logs to retrieve in ISO 8601 format",
+        ),
+    ],
+) -> list[dict[str, JsonValue]]:
+    logs = get_okta_logs(
+        query="policy.lifecycle.update Password",
+        start_time=start_time,
+        end_time=end_time,
+        secrets={"OKTA_SECRET": "okta_secret"},
+    )
+
+    return [
+        log
+        for log in logs
+        if log.get("target", [{}])[0].get("detailEntry", {}).get("policyType")
+        == "Password"
+    ]
+
+
+@action(
+    display_name="Format Okta Policy Update Message",
+    display_namespace="Okta",
+    description="Format Okta policy update logs into a readable message",
+)
+def format_okta_policy_update_message(
+    logs: Annotated[
+        list[dict[str, JsonValue]],
+        ArgumentMetadata(
+            display_name="Okta Logs",
+            description="List of Okta policy update logs",
+        ),
+    ],
+) -> str:
+    message = f"Attention: {len(logs)} Okta password policy update(s) detected in the last hour.\n\n"
+    for log in logs:
+        new_policy = json.loads(
+            log.get("debugContext", {})
+            .get("debugData", {})
+            .get("newPolicyExtensiblePropertiesJson", "{}")
+        )
+
+        old_policy = json.loads(
+            log.get("debugContext", {})
+            .get("debugData", {})
+            .get("oldPolicyExtensiblePropertiesJson", "{}")
+        )
```

Comment:
This is a bit cleaner imo:

```suggestion
        debug_data = log.get("debugContext", {}).get("debugData", {})

        new_policy = json.loads(
            debug_data.get("newPolicyExtensiblePropertiesJson", "{}")
        )
        old_policy = json.loads(
            debug_data.get("oldPolicyExtensiblePropertiesJson", "{}")
        )
```